app/helpers/cms/form_helper.rb

```diff
@@ -1,59 +1,29 @@
+# frozen_string_literal: true
+
 module CMS
   module FormHelper
 
     def cms_form_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
       options[:url] ||= polymorphic_path([:provider, :admin, template])
-      options[:as] ||= base_name
+      options[:as] ||= base_name(template)
 
       semantic_form_for(template, options, &block)
     end
 
     def cms_remote_form_for(template, options = {}, &block)
-      cms_form_for(template, options.reverse_merge(:html => { :"data-remote" => 'true' }), &block)
+      cms_form_for(template, options.reverse_merge({ remote: true }), &block)
     end
 
     def cms_fields_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
-      options[:as] ||= base_name
 
-      monkey_patched_semantic_fields_for(template, options, &block)
+      semantic_fields_for(base_name(template), template, options, &block)
```

Comment:
@nidhi-soni1104 This is the line that fixes the issue.

Originally the method (the monkey-patched one) received only the object itself, which in this case is `CMS::Page`, then the formtastic gem was in charge of generating the form fields, and it gave the inputs the name `cms_page`.

In this fix, I am passing first the `record_name` first, the value of which is `cms_template`, and then the CMS::Page **object**. In this case the generated form has `cms_template` in input names.

And as the controller expects the inputs under `cms_template` (and not `cms_page`) this fixes the issue.

Reply:
Okay Now I got it